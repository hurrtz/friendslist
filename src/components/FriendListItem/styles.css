:local(.friendListItem) {
  list-style: none;
  padding: 20px 10px 20px 20px;
  background-color: white;
  border-bottom: 1px solid #e3e3e3;
  display: flex;
  flex-direction: column;
  cursor: pointer;
}

:local(.friendListItem hr) {
  width: 100%;
}

:local(.contentSection) {
  display: flex;
}

:local(.isHidden) {
  display: none;
}

:local(.friendInfos) {
  flex: 1 0 auto;
}

:local(.friendInfos span) {
  font-weight: bold;
}

:local(.friendActions) {
  flex: 0 0 90px;
}

:local(.friendActions.align) {
  display: flex;
  align-items: center;
}

:local(.btnAction) {
  width: 40px;
}

:local(.btnAction.selected) {
  color: #333;
  background-color: #e6e6e6;
  border-color: #adadad;
}

:local(.btnAction),
:local(.btnAction):active,
:local(.btnAction):focus,
:local(.btnAction):hover {
  margin-left: 5px;
  color: #5c75b0;
}

button:focus {
  outline: 0 !important;
}
