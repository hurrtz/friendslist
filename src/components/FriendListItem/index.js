import React, { PureComponent } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import styles from './styles.css';

class FriendListItem extends PureComponent {
  render() {
    const { name, starFriend, starred, id, deleteFriend } = this.props;

    return (
      <li className={styles.friendListItem}>
        <div className={styles.friendInfos}>
          <div>
            <span>{name}</span>
          </div>
          <div>
            <small>xx friends in common</small>
          </div>
        </div>
        <div className={styles.friendActions}>
          <button
            className={`btn btn-default ${styles.btnAction}`}
            onClick={() => starFriend(id)}
          >
            <i
              className={classnames('fa', {
                'fa-star': starred,
                'fa-star-o': !starred,
              })}
            />
          </button>
          <button
            className={`btn btn-default ${styles.btnAction}`}
            onClick={() => deleteFriend(id)}
          >
            <i className="fa fa-trash" />
          </button>
        </div>
      </li>
    );
  }
}

FriendListItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  starred: PropTypes.bool,
  starFriend: PropTypes.func.isRequired,
};

export default FriendListItem;
