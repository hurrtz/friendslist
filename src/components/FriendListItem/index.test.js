import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

import FriendListItem from './';

Enzyme.configure({ adapter: new Adapter() });

const defaultProps = {
  id: 1,
  name: 'Name1',
  starred: true,
  starFriend: () => {},
  setSex: () => {},
};

function setup(injectedProps = {}) {
  const props = {
    ...defaultProps,
    ...injectedProps,
  };

  const enzymeWrapper = shallow(<FriendListItem {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

describe('FriendListItem', () => {
  it('should render self and subcomponents', () => {
    const { enzymeWrapper, props } = setup();

    expect(enzymeWrapper.find('li').exists()).toBe(true);
  });

  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <FriendListItem
          id={1}
          name="Name1"
          starred={true}
          starFriend={() => {}}
          setSex={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
