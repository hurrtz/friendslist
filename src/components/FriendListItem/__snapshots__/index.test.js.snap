exports[`FriendListItem should match the snapshot 1`] = `
<li
  className={undefined}
  onClick={[Function]}>
  <div
    className={undefined}>
    <div
      className={undefined}>
      <div>
        <span>
          Name1
        </span>
      </div>
      <div>
        <small>
          xx friends in common
        </small>
      </div>
    </div>
    <div
      className={undefined}>
      <button
        className="btn btn-default"
        onClick={[Function]}>
        <i
          className="fa fa-star" />
      </button>
      <button
        className="btn btn-default"
        onClick={[Function]}>
        <i
          className="fa fa-trash" />
      </button>
    </div>
  </div>
  <hr
    className="undefined" />
  <div
    className="undefined">
    <div
      className="">
      <small>
        Sex:
      </small>
      <button
        className="btn btn-default"
        onClick={[Function]}>
        <i
          className="fa fa-mars" />
      </button>
      <button
        className="btn btn-default"
        onClick={[Function]}>
        <i
          className="fa fa-venus" />
      </button>
    </div>
  </div>
</li>
`;
